### PR TITLE
migration: comment schema and validation

### PR DESCRIPTION
**Pull request by realaravinth**
JSON schema and validation for comments. Please see https://forum.forgefriends.org/t/common-json-schema-for-repository-information/563 for motivation

### CODE REVIEW

File: modules/migration/schemas/comment.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  "title": "Comment",
+  "description": "Comments received on issues associated to a repository within a forge (Gitea, GitLab, etc.).",
+
+  "type": "array",
+  "items": {
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "issue_index": {
+        "description": "Unique identifier, relative to the issue.",
```
Comment (singuliere) [inline modules/migration/schemas/comment.json]:
```suggestion
        "description": "Unique identifier, relative to the repository.",
```

Unless I'm mistaken the index is not relative to the issue but to the repository.

File: modules/migration/schemas/comment.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "poster_name": {
+        "description": "Name of the user who authored the comment.",
+        "type": "string"
+      },
+      "poster_email": {
+        "description": "Email of the user who authored the comment.",
+        "type": "string"
+      },
+      "content": {
+        "description": "Long, multiline, description.",
```
Comment (singuliere) [inline modules/migration/schemas/comment.json]:
```suggestion
        "description": "Multiline comment.",
```

**Review by singuliere — CHANGES_REQUESTED** [review]
Two minor documentation modifications and it is good to go. :+1:

File: modules/migration/schemas/comment.json (hunk 1 above)
Comment (realaravinth) [reply modules/migration/schemas/comment.json]: Thanks for the quick review!

I mistook the `issue_index` to be an identifier for the comment but upon further digging, I discovered that it identifies the issue. I will make the suggested change :+1:

File: modules/migration/schemas/comment.json (hunk 2 above)
Comment (realaravinth) [reply modules/migration/schemas/comment.json]: Good catch, thanks! I copied stuff from the issue schema for consistency not sure how I missed this :)

**Comment by realaravinth** [comment]
@lunny: if accommodating this PR before your #18165 is too much work, please feel free to merge it after your PR lands.

**Review by singuliere — APPROVED** [review]

**Comment by lunny** [comment]
Please resolve the conflict

**Comment by wxiaoguang** [comment]
Is this PR active?

**Comment by wxiaoguang** [comment]
It has been stale for a long time and I can't think of a way to handling it other than closing it. Feel free to reopen if there's any new progress and I could also help.

**Pull request closed without merging**